src/psyclone/transformations.py

```diff
@@ -1285,29 +1288,32 @@ class OMPSingleTrans(ParallelRegionTrans):
     '''
     # The types of node that this transformation cannot enclose
     excluded_node_types = (nodes.CodeBlock, nodes.Return, nodes.ACCDirective,
-                           psyGen.HaloExchange)
+                           psyGen.HaloExchange, nodes.OMPSingleDirective,
+                           nodes.OMPParallelDirective)
 
     def __init__(self, nowait=False):
         super(OMPSingleTrans, self).__init__()
         # Set the type of directive that the base class will use
         self._pdirective = self._directive
-        #Store whether this single directive has a barrier or not
+        # Store whether this single directive has a barrier or not
         self._omp_nowait = nowait
 
     def __str__(self):
         return "Insert an OpenMP Single region"
 
     @property
     def omp_nowait(self):
-        ''' Returns whether or not this Single region uses a nowait
-            clause to remove the end barrier. The default value is False
+        ''' :returns: whether or not this Single region uses a nowait \
+                      clause to remove the end barrier. The default value \
```

Comment:
No need to specify what the default value is here (since this is only describing the getter).

Reply:
Oki - fixed this and declared the parameter to the constructor in the class docs (which I'd missed before)

Reply:
Thanks - I missed that too. However, you've said it's of type `str` when it should be `bool`?